index.md

```diff
@@ -213,34 +213,34 @@ and give you credit for your work.
-    *   Modern file systems can handle hundreds of thousands of files in a directory,
-        but displaying contents is problematic
-5.  `results` for generated (intermediate) files
-    *   Most programmers frown on storing generated files (because you can regenerate them)
-    *   Researchers should so that they can easily tell if generated results have changed
-    *   Note that figures, tables, and other things expected to go into publications count as generated results
+Files should be placed into a main directory named for the project, and organized into a standard set of subdirectories following [William Noble's][noble-rules] recommendations:
+
+1.  `doc` contains text documents associated with the project. This may include files for manuscripts, documentation for source code, and/or an electronic lab notebook recording your experiments. Subdirectories may be created for these different classes of files.
+2.  `data` contains raw data and metadata, organized into subdirectories if needed. Note that this directory contains only unprocessed data, cleaned or otherwise modified data files are considered a result.
+3.  `src` contains the source code for scripts and programs, which may be written in interpreted languages such as R or Python or compiled languages like Fortran, C++, or Java.
+4.  `bin` contains executable scripts and programs that are brought in from other sources or compiled from code in the `src` directory. Projects that use only modern interpreted languages, such as R or Python, will not require this directory.
+5.  `results` for all files that are generated as part of the project. This includes both intermediate results, such as cleaned data sets or simulated data, as well as final results such as figures and tables.
 
 ### Discussion
 
-FIXME
+The `src` directory often contains two conceptually distinct types of files that should be distinguished either by clear file names or by additional subdirectories. The first type are individual files, or related groups of files, that contain functions to perform the core analysis of the research. If there are more than one or two of these files, these should be organized into additional subdirectories following language-specific rules or conventions (e.g., C++ header files, Python or R package structure).
+
+A second type of file in this directory are controller or driver scripts that combine the core analytical functions with specific analysis parameters and data input and output commands in order to execute the entire project analysis, from start to finish. A simple controller script, for example, may read in a raw data table, import and apply several analysis functions from the other files in this directory, and create and save a figure. For a small project with one main output, a single controller script should be placed in the main `src` directory and distinguished clearly from the other scientific code by a name such as "runall".
```

Comment:
I wouldn't go that far, as we're not discussing paths or package conventions in any detail, which would be more in line with reuse of code across projects. A closer idea would be that the stuff in the function files or modules is what, if we were recommending unit testing, would be unit tested, while the runall should just be the glue, completely transparent and obvious as to what it's doing. Another way to think of it is that, in simple projects, the `runall` is where the analysis functions actually meet the data (so it's where code and data come together). I'll clarify in revision.